### PR TITLE
fix: remove epic-spinners

### LINKED ISSUE #188: Remove epic-spinners from vuestic-ui
Rationale:

* Is used only for loaders.
* Breaks tests (we want tests to be part of CI like right now)

Currently it's used to show spinner on loading in select and data table. We should just use icon instead. You may add `spin` prop to va-icon for this to work.

### PR DESCRIPTION
**Pull request by bdrtsky**
closes #188

## Description
Remove epic-spinners

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: src/components/vuestic-components/va-icon/VaIcon.vue [hunk 1]
```diff
@@ -85,7 +89,7 @@ export default {
       }
     },
     computedContent () {
-      return this.icon && this.icon.content
+      return this.icon ? this.icon.content : this.spin ? 'refresh' : ''
```
Comment (asvae) [inline src/components/vuestic-components/va-icon/VaIcon.vue]: Spin prop should function similarly to `fa-spin`. It shouldn't modify existing icon.

File: src/components/vuestic-components/va-icon/VaIcon.vue [hunk 2]
```diff
@@ -95,5 +99,19 @@ export default {
 .va-icon {
   vertical-align: middle;
   user-select: none;
+
+  &.spin {
```
Comment (asvae) [inline src/components/vuestic-components/va-icon/VaIcon.vue]: Bem.

File: src/components/vuestic-components/va-icon/VaIcon.vue [hunk 3]
```diff
@@ -95,5 +99,19 @@ export default {
...
+    animation: rotate 1000ms infinite;
+  }
+}
+
+@keyframes rotate {
```
Comment (asvae) [inline src/components/vuestic-components/va-icon/VaIcon.vue]: Please use something similar to [this solution](https://github.com/epicmaxco/epic-spinners/blob/master/src/services/utils.js).
We ideally do not want to introduce any global css.

File: src/components/vuestic-components/va-icon/VaIcon.vue [hunk 4]
```diff
@@ -46,6 +46,10 @@ const IconContextMixin = makeContextablePropsMixin({
     type: [String, Number],
     default: '',
   },
+  spin: {
```
Comment (asvae) [inline src/components/vuestic-components/va-icon/VaIcon.vue]: All props should be represented in demo.

Please also add all icontypes as well as svg/component icons to card to ensure spinning works properly in all cases.

File: src/components/vuestic-components/va-icon/VaIcon.vue [hunk 5]
```diff
@@ -59,7 +63,7 @@ export default {
       return (this.icon && this.icon.component) || this.c_component || this.c_tag
     },
     computedClass () {
-      return (this.icon && this.icon.iconClass) || ''
+      return this.icon ? this.icon.iconClass : this.spin ? 'material-icons spin' : ''
```
Comment (asvae) [inline src/components/vuestic-components/va-icon/VaIcon.vue]: Why would you add `material-icons` here? This should work with any iconfont.

File: src/components/vuestic-components/va-icon/VaIcon.vue (hunk 4 above)
Comment (bdrtsky) [reply src/components/vuestic-components/va-icon/VaIcon.vue]: What is `icontypes`?

Comment (asvae) [reply src/components/vuestic-components/va-icon/VaIcon.vue]: Meant iconfont, sorry.

File: src/components/vuestic-components/va-icon/VaIcon.vue (hunk 1 above)
Comment (bdrtsky) [reply src/components/vuestic-components/va-icon/VaIcon.vue]: Material icons working by adding text in `i` tag, not CSS classes

File: src/components/vuestic-components/va-icon/VaIcon.vue (hunk 5 above)
Comment (bdrtsky) [reply src/components/vuestic-components/va-icon/VaIcon.vue]: Different icon fonts working differently. Awesome working by adding CSS class, Material - by adding text in `i`. How I suppose to detect which font is used?

File: src/components/vuestic-components/va-icon/VaIcon.vue (hunk 4 above)
Comment (bdrtsky) [reply src/components/vuestic-components/va-icon/VaIcon.vue]: Still not clear what `all cases` are meant? What `all cases`?

**Comment by asvae** [comment]
One notable miss was that animation didn't have linear flag on it (when fa-spin did).

I fixed that, and also a couple of minor things.

**Pull request merged**